src/main/java/filter/expression/Qualifier.java

```diff
@@ -117,6 +154,31 @@ public static boolean process(IModel model, FilterExpression expr, TurboIssue is
         return exprWithNormalQualifiers.isSatisfiedBy(model, issue, new MetaQualifierInfo(metaQualifiers));
     }
 
+    private static int getCurrentMilestoneIndex(List<TurboMilestone> allMilestones) {
+        int currentIndex = -1;
+
+        for (TurboMilestone checker : allMilestones) {
+            // current milestone must be open
+            if (checker.isOpen()) {
+                // if milestone is overdue but have no open issue, ignore
+                if (checker.getDueDate().isPresent() &&
+                        checker.getDueDate().get().isBefore(LocalDate.now()) &&
+                        checker.getOpenIssues() == 0) {
+                    continue;
+                }
+                currentIndex = allMilestones.indexOf(checker);
+                break;
```

Comment:
I will still need the continue as otherwise the eligible boolean will be very long (listing 1 negative vs listing 3 positives). I will refactor it to like what you suggested to be cleaner.